spring-amqp/src/main/java/org/springframework/amqp/support/converter/DefaultClassMapper.java

```diff
@@ -186,4 +199,48 @@ public void fromClass(Class<?> clazz, MessageProperties properties) {
 		return toClass(classId);
 	}
 
+	private Class<?> toClass(String classId) {
+		if (this.idClassMapping.containsKey(classId)) {
+			return this.idClassMapping.get(classId);
+		}
+		if (classId.equals(DEFAULT_HASHTABLE_TYPE_ID)) {
+			return this.defaultMapClass;
+		}
+		try {
+			Class<?> aClass = ClassUtils.forName(classId, getClass().getClassLoader());
```

Comment:
Do you think it's safe to instantiate the class? In Kafka I compared the packages as String.